modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/InjectableParameterResolver.java

```diff
@@ -65,25 +65,18 @@ public InjectableParameterResolver(ParameterizedModel parameterizedModel,
    * @return the value of the injectable parameter.
    */
   public Object getInjectableParameterValue(String parameterName) {
-    Object parameterValue;
-    parameterValue = resolvedParameterValues.get(parameterName);
+    Object parameterValue = null;
     InjectableParameterInfo injectableParameterInfo = injectableParametersMap.get(parameterName);
-
-    if (parameterValue != null) {
-      try {
-        parameterValue = expressionManager
-            .evaluate("#[payload]",
-                      DataType.fromType(getClassFromType(injectableParameterInfo.getType())),
-                      BindingContext.builder()
-                          .addBinding("payload", new TypedValue(parameterValue, DataType.fromObject(parameterValue))).build())
-            .getValue();
-      } catch (ClassNotFoundException e) {
-        LOGGER.debug("Transformation of injectable parameter '{}' failed, the same value of the resolution will be used.",
-                     parameterName);
-      }
-
+    try {
+      parameterValue = expressionManager
+          .evaluate("#[ payload." + parameterName + "]",
+                    DataType.fromType(getType(injectableParameterInfo.getType())),
+                    expressionResolvingContext)
+          .getValue();
+    } catch (IllegalArgumentException e) {
```

Comment:
shouldn't this just catch any exception?

Reply:
I just tried to mimic the previous behavior. Before we only catched a ClassNotFoundException and returned null in that case. But any other exception was re thrown. 
But you are right that this IllegalArgumentException could make no sense.